subsystem/src/subsystem.rs

```diff
@@ -71,6 +57,19 @@ impl<T: 'static> CallRequest<T> {
     }
 }
 
+/// Call response that can be polled for result
+pub struct CallResponse<T>(oneshot::Receiver<T>);
+
+impl<T> std::future::Future for CallResponse<T> {
+    type Output = Result<T, CallError>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> task::Poll<Self::Output> {
+        Pin::new(&mut self.0)
+            .poll(cx)
+            .map(|r| r.map_err(|_| CallError::ResultFetchFailed))
```

Comment:
Nitpick: `Poll` provides the `map_err` function, so it can be used directly:
```suggestion
        Pin::new(&mut self.0).poll(cx).map_err(|_| CallError::ResultFetchFailed)
```

Reply:
Very nice, cheers!